### DISCUSSION TITLE
How can I upload generated PDF ?

### DISCUSSION THREAD

**Discussion by PierroD** [py-pdf/fpdf2#342] [Q&A]
Hi :smile: 
I'm looking for a way to upload the pdf I just built directly to my s3 bucket.

:warning: I don't want to "output" the file

:heavy_check_mark: I would like to use my fpdf 'object' to upload it directly to my bucket

:question:  Is there anyway to do it without generating a local pdf file ?

Thanks :smile:

**Comment by Lucas-C** [comment]
I think you should be able to use [the Python `boto` client for AWS](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/s3.html#client). The [`S3.Client.put_object`](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/s3.html#S3.Client.put_object) allows to send `bytes`, which is the output of `FPDF.output()`. I'm sure you can find tutorials online to help you with using `boto` & S3 😊

**Reply by PierroD** [reply]
I'm currently using boto, I will try to send it using output("I") which give me a string of bytes if I correctly remember

**Reply by Lucas-C** [reply]
While it may have worked with `pyfpdf`, with `fpdf2` `output("I")` will save the PDF under a file named `I`, which is not what you want.
Check out the module documentation: https://pyfpdf.github.io/fpdf2/fpdf/fpdf.html#fpdf.fpdf.FPDF.output
